.kokoro/deps-finder.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#Compare shared-dependencies version in a given dependency to desired version.
+#Return 0 for success, 1 for failed to find POM, 2 for shared-deps not present, 3 for incorrect version.
+
+#Options to run:
+#0 Arguments: Test the latest commit (with a dependency update) for the latest version of shared-deps.
+#1 Argument: Input artifactId:version to test the given dependency for the latest version of shared-deps.
+#2 Arguments: Input artifactId:version to test the given dependency for a specific version of shared-dependencies
+#(only input a version number for the second argument).
+
+## Get the directory of the build script
+scriptDir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+
+# include common functions
+source ${scriptDir}/common.sh
+
+#Returns 1 if we find an updated dependency in the commit message.
+function getDependency() {
+  currCommitNumber=$(git rev-parse HEAD)
+
+  #Note that this grep serves to check if we're updating a dependency including shared-dependencies.
+  currCommitMessage=$(git log -1 $currCommitNumber | grep "deps: update dependency com.google.cloud:google-cloud-")
```

Comment:
Let's be consistent with our variable references (i.e. always use the curly brackets for the variable names):
```suggestion
  currCommitMessage=$(git log -1 ${currCommitNumber} | grep "deps: update dependency com.google.cloud:google-cloud-")
```